### PR TITLE
Manage Impact message text translation in different APIs

### PR DESCRIPTION
**Pull request by kadhikari**
Manage Impact message text translation if present in chaos disruption for different APIs. If Translation for a language_value (&parameter=language_value) is present in the message then replace the message text by message.translation.text. The followings are the APIs:
- journeys
- line_reports
- disruptions
- traffic_reports

chaos-proto: https://github.com/hove-io/chaos-proto/blob/fbf19481022fc0384768993615a4303ca65691c1/chaos.proto#L197
navitia-proto: https://github.com/hove-io/navitia-proto/blob/e82017345a8710fce0d0e56edb67ade88e1a8caf/request.proto#L286

For more details: https://navitia.atlassian.net/browse/NAV-2605

### CODE REVIEW

File: source/type/pb_converter.cpp [hunk 1]
```diff
@@ -1343,6 +1343,12 @@ void PbCreator::Filler::fill_pb_object(const nd::Impact* impact, pbnavitia::Impa
     for (const auto& m : impact->messages) {
         auto pb_m = pb_impact->add_messages();
         pb_m->set_text(m.text);
+        const auto language = this->pb_creator.language;
```
Comment (pbench) [inline source/type/pb_converter.cpp]:
```suggestion
```
I think you don't use the variable `language`

File: source/type/pb_converter.cpp [hunk 2]
```diff
@@ -1849,6 +1855,16 @@ std::string PbCreator::get_section_id(pbnavitia::Journey* j, size_t section_idx)
     return it->second;
 }
 
+std::string PbCreator::get_translated_message(std::vector<type::disruption::Translation> translations,
```
Comment (pbench) [inline source/type/pb_converter.cpp]:
```suggestion
std::string PbCreator::get_translated_message(std::vector<type::disruption::Translation> const & translations,
```
You don't want to copy the vector when calling this function !

File: source/ed/build_helper.cpp [hunk 3]
```diff
@@ -544,16 +544,25 @@ Impacter& Impacter::msg(dis::Message m) {
...
-    m.channel_id = str;
     m.channel_name = str + " channel";
     m.channel_content_type = "content type";
     m.created_at = boost::posix_time::ptime(b.data->meta->production_date.begin(), boost::posix_time::minutes(0));
 
+    if (!translated_text.empty()) {
+        dis::Translation t;
+        t.text = translated_text;
+        t.language = language;
+        m.translations.push_back(std::move(t));
```
Comment (xlqian) [inline source/ed/build_helper.cpp:563]:
```suggestion
        m.translations.emplace_back(translated_text, language);
```

File: source/kraken/make_disruption_from_chaos.cpp [hunk 4]
```diff
@@ -280,6 +280,15 @@ static std::vector<nt::disruption::PtObj> make_pt_objects(
     return res;
 }
 
+static std::vector<nt::disruption::Translation> make_translations(
+    const google::protobuf::RepeatedPtrField<chaos::Translation>& chaos_translations) {
+    std::vector<nt::disruption::Translation> res;
+    for (const auto& chaos_trans : chaos_translations) {
+        res.push_back({chaos_trans.text(), chaos_trans.language(), chaos_trans.url_audio()});
```
Comment (xlqian) [inline source/kraken/make_disruption_from_chaos.cpp:287]:
```suggestion
        res.emplace_back(chaos_trans.text(), chaos_trans.language(), chaos_trans.url_audio());
```

**Review by xlqian — APPROVED** [review]

**Review by pbench — APPROVED** [review]

**Pull request merged**